lib/codecs/tests/data/encoding/protobuf/test.proto

```diff
@@ -0,0 +1,32 @@
+syntax = "proto3";
+
+package test;
+
+message Integers {
+  int32 i32 = 1;
+  int64 i64 = 2;
+  uint32 u32 = 3;
+  uint64 u64 = 4;
+}
+
+message Floats {
+  double d = 1;
+  float f = 2;
+}
+
+message Bytes {
+  string text = 1;
+  bytes binary = 2;
+}
+
+message RepeatedPrimitive {
+  repeated int64 numbers = 1;
+}
+
+message RepeatedMessage {
+  message EmbeddedMessage {
+    optional string text = 1;
+    optional uint32 index = 2;
+  }
+  repeated EmbeddedMessage messages = 1;
+}
```

Comment:
Let's also add an `enum Foo`, a `default` and a `map` similar to `lib/codecs/tests/data/decoding/protobuf/test_protobuf3.proto`.

Reply:
Hahaha, guess who forgot to finish implementing all the protobuf constructs?  Added `enum`, `map`, and `google.protobuf.Timestamp`.

`default` is interesting because it's removed in `proto3` (only `proto2` allows custom default values).  For encoding, this shouldn't matter anyway, since `prost` will load the default when it parses the `.desc` file and use that default if we don't explicitly set a field during the encoding process.  Thoughts on if we should even test such a situation?

Reply:
Thank you for strengthening the tests.

Hmm, we stumbled upon the `no presence` (proto3) vs `explicit presence` (proto2) semantics here. We might have a situation where we cannot differentiate between a non-set value and a default value. 

However, we have `optional` in proto3 as a workaround. The type would be `Option<T>` for such fields as described in the [docs](https://github.com/tokio-rs/prost#field-modifiers). Repeated fields are an interesting case here since they are just `Vec<T>` and not `Option<Vec<T>>`.

Reply:
I'm not 100% sure how this maps to the encoding implementation.  The implementation here only attempts to set fields that are part of the vector `Value::Object`, meaning that prost itself is left to decide what happens when the value is unset - it will either encode it as missing or encode it with a default value.  If the field is `optional`, that could change how prost encodes the data, but shouldn't change how we set the fields, no?